draft-lenders-dns-cbor.md

```diff
@@ -627,32 +558,123 @@ dns-response = [
 ~~~
 {:cddl #fig:dns-response title="DNS Response Definition"}
 
-# Further Compression with CBOR-packed {#sec:cbor-packed}
+# Compression with CBOR-packed {#sec:cbor-packed}
+
+CBOR-packed {{-cbor-packed}} is used for name compression in application/dns+cbor.
 
-If both DNS server and client support CBOR-packed {{-cbor-packed}}, it MAY be used for further
+If both DNS server and client support table setup tag 113 as described in {{Section 3.1 of -cbor-packed}}, it MAY be used for further
 compression in DNS responses.
 Especially IPv6 addresses, e.g., in AAAA resource records can benefit from straight referencing to
 compress common address prefixes.
 
-## Media Type Negotiation
+## Name Compression {#sec:name-compression}
 
-A DNS client uses the media type "application/dns+cbor;packed=1" to negotiate (see, e.g.,
-{{-http-semantics}} or {{-coap}}, Section 5.5.4) with the DNS server whether the server supports packed
-CBOR.
-If it does, it MAY request the response to be in CBOR-packed (media type
-"application/dns+cbor;packed=1").
-The server then SHOULD reply with the response in CBOR-packed, which it also signals with media type
-"application/dns+cbor;packed=1".
+~~~ cddl
+Text-String-Suffix-Sequence-Packed-CBOR = #6.28259(rump)
+~~~
+
+For name compression, a new packing table setup tag TBD28259 ('n' and 'c' in ASCII) for CBOR-packed {{-cbor-packed}} is defined.
+It provides an implicit text string suffix sequence table for shared items _V_ which is appended to the existing table for shared items of any table setup tag within the content of tag TBD28259 (by default empty table).
+This implicit (i.e. not explicitly represented) table _V_ is constructed as follows:
+Any coherent sequence of text strings encountered within the rump of tag TBD28259, as well as any of its non-empty suffixes, are added to the table as arrays in depth-first order.
+Text string sequences within any tables for shared items or argument items within the rump MUST not be added to _V_.
+If a sequence for which an array is already in _V_ is encountered, a shared item reference _i_ to that array in V replaces this sequence.
+This shared item reference _i_ means: take the sequence from the array at _V_\[_i_\] and put it into the surrounding array in place of _i_.
+The resulting rump should look like referencing the _i_-th string (depth first) in the sequence.
 
-## DNS Representation in CBOR-packed
+The "application/dns+cbor" media type comes with an optional parameter "packed".
+If it is not provided, the value of it, i.e. the level of packedness, is assumed to be 0.
```

Comment:
I am not set on the term “level of packedness”. In fact, I would love to get some alternative terms.